source/server/config/http/lightstep_http_tracer.cc

```diff
@@ -0,0 +1,44 @@
+#include "server/config/http/lightstep_http_tracer.h"
+
+#include <string>
+
+#include "common/tracing/http_tracer_impl.h"
+#include "common/tracing/lightstep_tracer_impl.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+Tracing::HttpTracerPtr LightstepTracerFactory::tryCreateHttpTracer(const std::string &type,
+                                                const Json::Object& json_config,
+                                                Server::Instance& server,
+                                                Upstream::ClusterManager &cluster_manager) {
+  if (type != "lightstep") {
```

Comment:
Yes, I was following how the filters were done.  I would prefer a map approach, too, because it's not ordering dependent (you can error if there are two implementations that match rather than just choosing the one that statically registered first).